packages/FormElement/src/FormElement.js

```diff
@@ -0,0 +1,164 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import isNil from "lodash/isNil";
+import isString from "lodash/isString";
+import uuid from "uuid/v1";
+
+import { extractChildren } from "./helpers/extractChildren";
+import Description from "./components/Description";
+import ErrorMessage from "./components/ErrorMessage";
+import Help from "./components/Help";
+import Label from "./components/Label";
+
+import formElementStyles, { inlineContainerStyles } from "./FormElement.styles";
+
+const propTypes = {
+  children: PropTypes.node.isRequired,
+
+  /** Should show is optional text besides the label or not. */
+  hasOptionalLabel: PropTypes.bool,
+
+  /** Should show is required text besides the label or not. */
+  hasRequiredLabel: PropTypes.bool,
+
+  /** ID for the child element. */
+  id: PropTypes.string,
+
+  /** Should be disabled or not, default is false. */
+  isDisabled: PropTypes.bool,
+
+  /** Should label and children be inline or not, default is false. */
+  isInline: PropTypes.bool,
+
+  /** Should label be hidden, default is false. Note: this is discouraged because of accessibility requirements. */
+  isLabelVisuallyHidden: PropTypes.bool,
+
+  /** Should be read-only or not, default is false. */
+  isReadOnly: PropTypes.bool,
+
+  /** Label text of this field. */
+  label: PropTypes.string.isRequired,
+
+  /** Size of the label, child component, error, help and description (font size, min-height, padding, etc). */
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  hasOptionalLabel: false,
+  hasRequiredLabel: false,
+  id: null,
+  isDisabled: false,
+  isInline: false,
+  isLabelVisuallyHidden: false,
+  isReadOnly: false,
+  size: "medium",
+};
+
+function FormElement(props) {
+  const {
+    children,
+    hasOptionalLabel,
+    hasRequiredLabel,
+    id,
+    isDisabled,
+    isInline,
+    isLabelVisuallyHidden,
+    isReadOnly,
+    label,
+    size,
+  } = props;
+
+  const extratedChildren = extractChildren(children, [
+    "FormElement.Description",
+    "FormElement.Error",
+    "FormElement.Help",
+  ]);
+  const ariaDescriptionId = uuid();
+  const hasError = !!extratedChildren["FormElement.Error"] && !!extratedChildren["FormElement.Error"].props.children;
+  const uniqueId = isNil(id) || id === "" ? uuid() : id;
```

Comment:
This will created a new uuid each render.

`const uuid = React.useRef(uuidV4());`

would work better.